spec/lib/replica_spec.rb

```diff
@@ -172,9 +137,38 @@
         rev_start = 2015_02_12_003430
         rev_end = 2015_03_10_003430
 
-        response = Replica.get_revisions(all_users, rev_start, rev_end, 'es')
+        es_wiki = Wiki.new(language: 'es', project: 'wikipedia')
+        response = Replica.new(es_wiki).get_revisions(all_users, rev_start, rev_end)
         expect(response.count).to eq(24)
       end
     end
+
+    it 'handles timeout errors' do
+      stub_request(:any, %r{http://tools.wmflabs.org/.*})
+        .to_raise(Errno::ETIMEDOUT)
+      all_users = [
+        build(:user, username: 'ELE427', id: 22905965),
+        build(:user, username: 'Ragesoss', id: 319203),
+        build(:user, username: 'Mrbauer1234', id: 23011474)
+      ]
+      rev_start = 2014_01_01_003430
+      rev_end = 2014_12_31_003430
+
+      response = Replica.new.get_revisions(all_users, rev_start, rev_end)
+      expect(response).to be_empty
+    end
+
+    it 'handles connection refused errors' do
+      stub_request(:any, %r{http://tools.wmflabs.org/.*})
+        .to_raise(Errno::ECONNREFUSED)
+      all_users = [
```

Comment:
Similar code found in replica_spec.rb:8, replica_spec.rb:131, replica_spec.rb:149, replica_spec.rb:164